src/adagrams.js

```diff
@@ -1,15 +1,143 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+  let hand = [];
+  let randomLetter = "";
+  while (hand.length < 10) {
+    randomLetter = getRandomLetter(letterPool);
+    if (letterPool[randomLetter] > 0)  {
+      letterPool[randomLetter] -= 1;
+            hand.push(randomLetter.toUpperCase());
+    }
+  } 
+  return hand;
 };
 
+// ~~~~WAVE 2~~~~ //
+
+function getRandomLetter() {
+  const letters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+  const randomIndex = Math.floor(Math.random() * letters.length);
+  return letters[randomIndex];
+};
+
+// ~~~~WAVE 3~~~~ //
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  const bank = [...lettersInHand];
```

Comment:
Great job using the spread operator to make a shallow copy of lettersInHand! A shallow copy is a great idea here because lettersInHand is just a list of values as opposed to objects!